### PR TITLE
Support binding specified IP

### PR DESCRIPTION
**Pull request by xlch88**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -24,7 +24,8 @@ import (
 
 func main() {
 	var serverAddr = flag.String("s", stun.DefaultServerAddr, "STUN server address")
-	var localPort = flag.Int("p", 0, "The port on which to bind requests, set to 0 to pick a random port")
+	var localPort = flag.Int("p", 0, "The ip on which to bind requests, set to empty will keep default")
+	var localIP = flag.String("i", "", "")
```
Comment (AudriusButkevicius) [inline main.go:28]: Why not have one addreess that takes an address with a port?

File: stun/client.go [hunk 2]
```diff
@@ -98,12 +104,22 @@ func (c *Client) Discover() (NATType, *Host, error) {
...
 	if conn == nil {
 		var laddr *net.UDPAddr
+
+		address := ""
+		if c.localIP != "" {
+			address += c.localIP
+		}
 		if c.localPort != 0 {
-			laddr, err = net.ResolveUDPAddr("udp", fmt.Sprintf(":%d", c.localPort))
+			address += strconv.Itoa(c.localPort)
```
Comment (ccding) [inline stun/client.go:113]: i think we need a `:` between ip and port

File: stun/client.go [hunk 3]
```diff
@@ -98,12 +104,22 @@ func (c *Client) Discover() (NATType, *Host, error) {
...
+		}
+
+		if c.localPort != 0 {
```
Comment (ccding) [inline stun/client.go:116]: why do we not put the two `if c.localPort != 0 {` together?

File: main.go (hunk 1 above)
Comment (xlch88) [reply main.go:28]: Because in some cases you only need to specify the port, not the IP?

File: stun/client.go (hunk 2 above)
Comment (xlch88) [reply stun/client.go]: Yes, you're right, I forgot about this

File: stun/client.go (hunk 3 above)
Comment (xlch88) [reply stun/client.go]: I've taken care of it and look better now

File: stun/client.go [hunk 4]
```diff
@@ -98,12 +104,18 @@ func (c *Client) Discover() (NATType, *Host, error) {
 	conn := c.conn
 	if conn == nil {
 		var laddr *net.UDPAddr
-		if c.localPort != 0 {
-			laddr, err = net.ResolveUDPAddr("udp", fmt.Sprintf(":%d", c.localPort))
+
+		if c.localPort != 0  || c.localIP != "" {
+			var address = c.localIP + ":" + strconv.Itoa(c.localPort)
```
Comment (ccding) [inline stun/client.go:109]:
```suggestion
			var address = fmt.Sprintf("%s:%d", c.localIP, c.localPort)
```

File: stun/client.go [hunk 5]
```diff
@@ -127,11 +139,15 @@ func (c *Client) BehaviorTest() (*NATBehavior, error) {
 	conn := c.conn
 	if conn == nil {
 		var laddr *net.UDPAddr
-		if c.localPort != 0 {
-			laddr, err = net.ResolveUDPAddr("udp", fmt.Sprintf(":%d", c.localPort))
+		if c.localPort != 0  || c.localIP != "" {
+			var address = c.localIP + ":" + strconv.Itoa(c.localPort)
```
Comment (ccding) [inline stun/client.go:143]: same here

**Comment by ccding** [comment]
lgtm after fixing the above minor issues

File: main.go (hunk 1 above)
Comment (AudriusButkevicius) [reply main.go]: You could still do "192.168.0.10:0" or ":5738" or both etc.

**Comment by AudriusButkevicius** [comment]
What about ipv6 addresses?

**Comment by ccding** [comment]
ipv6 doesn't have NAT

**Comment by xlch88** [comment]
merge ?

**Pull request merged**